File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.rb
# @param {String} s
# @return {Integer}
def balanced_string_split(s)
    counter = 0
    ret_val = 0
    letter = s[0]
    s.each_char do |char|
        if char == letter
            counter += 1
        else
            counter += -1
        end
        if counter == 0
            ret_val += 1
        end
    end
    ret_val
end

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        ret_val = 0
        counter = 0
        first = s[0]
        for char in s:
            if char == first:
                counter += 1
            else:
                counter += -1
            if counter == 0:
                ret_val += 1
        return ret_val